pkg/vm/vm.go

```diff
@@ -1214,15 +1214,13 @@ func (v *VM) execute(ctx *Context, op opcode.Opcode, parameter []byte) (err erro
 			panic("invalid length")
 		}
 
-		items := make([]stackitem.MapElement, n)
-		for i := range n {
-			key := v.estack.Pop()
-			validateMapKey(key)
+		m := stackitem.NewMap()
+		for range n {
+			key := v.estack.Pop().value
 			val := v.estack.Pop().value
-			items[i].Key = key.value
-			items[i].Value = val
+			m.Add(key, val)
```

Comment:
The case of nil key is not handled by `m.Add`, the behaviour should be compatible with `validateMapKey`. And BTW, this change also fixes another bug that makes stack incompatible with C# node in case of invalid key, because C# node firstly POPs *both* key and value, and only after that validates map key.